internal/app/client/client.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2023 The Connect Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"flag"
+	"io"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"connectrpc.com/conformance/internal/app"
+	"connectrpc.com/conformance/internal/gen/proto/connect/connectrpc/conformance/v1alpha1/conformancev1alpha1connect"
+	v1alpha1 "connectrpc.com/conformance/internal/gen/proto/go/connectrpc/conformance/v1alpha1"
+	"connectrpc.com/connect"
+	"golang.org/x/net/http2"
+)
+
+// Wrapper defines a wrapper around a client which can invoke requests based on a ClientCompatRequest.
+type Wrapper interface {
```

Comment:
This doesn't need to be exported. In fact, I don't think you need an interface at all since there's only one implementation (`conformanceClientWrapper`) -- so callers could just refer directly to that one concrete type.

Reply:
Updated. I figured maybe this would be helpful if we ever need additional implementations?

Reply:
We can always change type signatures to use an interface in the future when we get there. Though, TBH, I can't think of anything that would necessitate a second implementation off the top of my head.

And since none of this importable library code, we don't have to be concerned about changing signatures and breaking users. So we can freely change signatures and types in the future if need be.